contracts/EmergencyPriceFeed.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: MIT License
+pragma solidity 0.7.6;
+pragma experimental ABIEncoderV2;
+
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { IUniswapV3Pool } from "@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol";
+import { FixedPoint96 } from "@uniswap/v3-core/contracts/libraries/FixedPoint96.sol";
+import { FullMath } from "@uniswap/v3-core/contracts/libraries/FullMath.sol";
+import { TickMath } from "@uniswap/v3-core/contracts/libraries/TickMath.sol";
+import { IPriceFeed } from "./interface/IPriceFeed.sol";
+import { BlockContext } from "./base/BlockContext.sol";
+
+contract EmergencyPriceFeed is IPriceFeed, BlockContext {
+    using Address for address;
+
+    //
+    // STATE
+    //
+
+    address public pool;
+
+    //
+    // EXTERNAL NON-VIEW
+    //
+
+    constructor(address poolArg) {
+        // EPF_EANC: pool address is not contract
+        require(address(poolArg).isContract(), "EPF_EANC");
+
+        pool = poolArg;
+    }
+
+    //
+    // EXTERNAL VIEW
+    //
+
+    function getPrice(uint256 interval) external view override returns (uint256) {
+        uint256 markTwapX96 = _formatSqrtPriceX96ToPriceX96(_getSqrtMarkTwapX96(_toUint32(interval)));
+        return _formatX96ToX10_18(markTwapX96);
+    }
+
+    //
+    // EXTERNAL PURE
+    //
+
+    function decimals() external pure override returns (uint8) {
+        return 18;
+    }
+
+    /// @dev if twapInterval < 10 (should be less than 1 block), return mark price without twap directly,
+    ///      as twapInterval is too short and makes getting twap over such a short period meaningless
+    function _getSqrtMarkTwapX96(uint32 twapInterval) internal view returns (uint160) {
```

Comment:
This function is copied from `UniswapV3Broker.getSqrtMarkTwapX96` in perp-lushan.
https://github.com/perpetual-protocol/perp-lushan/blob/release/v1.0.x/contracts/lib/UniswapV3Broker.sol#L226